test/steps/http.go

```diff
@@ -45,27 +60,39 @@ func (s *StepContext) SetAuthHeaders() {
 	s.authHeaders = "Bearer IAmAuthorized"
 }
 
-func (s *StepContext) getDefaultUploadRequestPayload() string {
-	return `{
-		"url": "http://localhost/input",
+func (s *StepContext) getDefaultUploadRequestPayload(sourceFilename string) string {
+	return fmt.Sprintf(`{
+		"url": "%s",
 		"callback_url": "http://localhost:3000/cb",
 		"output_locations": [
 			{
 				"type": "object_store",
 				"url": "memory://localhost/output.m3u8",
  				"outputs": {
-					"source_segments": true
+					"source_segments": true,
+					"transcoded_segments": true
 				}
-			},
+			}
+		]
+	}`, sourceFilename)
+}
+
+func (s *StepContext) getCustomSegmentSizeUploadRequestPayload(sourceFilename string) string {
```

Comment:
maybe we should make use of the struct definitions, build a struct with whatever optional fields we need then marshal to json?

Reply:
Yeah good call, I'd convinced myself that it wasn't worth it until we need to change one more thing but probably best to just get it done now.